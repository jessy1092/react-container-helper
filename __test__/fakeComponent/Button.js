
import React from 'react';

import { contain } from '../../src';

const Button = ({ toggle, handleClick }) => (
  <button onClick={handleClick}>
    {toggle ? 'true' : 'false'}
  </button>
);

const initState = () => ({
  toggle: false,
});

const mapStateToProps = ({ toggle }) => ({
  toggle,
});

const mapSetStateToProps = (setState, { toggle }) => ({
  handleClick: () => setState({ toggle: !toggle }),
});

export const ToggleButton = contain(initState, mapStateToProps, mapSetStateToProps)(Button);

export default Button;
