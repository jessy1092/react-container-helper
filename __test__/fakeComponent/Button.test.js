
import React    from 'react';
import renderer from 'react-test-renderer';
import { ToggleButton } from './Button';

test('Button changes state when clicked', () => {
  const component = renderer.create(
    <ToggleButton />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
