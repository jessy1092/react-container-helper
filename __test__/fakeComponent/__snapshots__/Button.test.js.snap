exports[`test Button changes state when clicked 1`] = `
<button
  onClick={[Function]}>
  false
</button>
`;

exports[`test Button changes state when clicked 2`] = `
<button
  onClick={[Function]}>
  true
</button>
`;
